Vampirism web api

Cheaterpaul opened the pull request:
- swap contributor api
- allows to enable/disable features remotely in the api
- sends basic telemetry data, if it is not disabled in minecraft and vampirism
  - mod version
  - minecraft version
  - installed mods
  - server/client
- automatic update of config values and contributors if pushed to the default branch

Cheaterpaul committed:
swap to settings api

Cheaterpaul committed:
remove custom version checker

Cheaterpaul committed:
add telemetry data

Cheaterpaul committed:
clean up REFERENCE file

maxanier commented on src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java:
```diff
@@ -0,0 +1,143 @@
+        return Optional.ofNullable(this.settingValues.get(key));
+    }
+
+    public boolean isSettingTrue(String key) {
```
Maybe it would be better to have a "default" parameter. Or just get rid of the nullable methods completely and only have optionals. (Optional<String> and Optional<Boolean> and Optional<Integer>)

maxanier commented on src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java:
```diff
@@ -0,0 +1,143 @@
+        return get("supporter/list").thenApply(x -> GSON.fromJson(x, TypeToken.getParameterized(List.class, Supporter.class).getType()));
+    }
+
+    public CompletableFuture<Collection<Supporter>> GetSupportersAsync(String modid) {
```
Upper case function name?

maxanier commented on src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java:
```diff
@@ -0,0 +1,143 @@
+        return get(this.apiVersion, path);
+    }
+
+    private CompletableFuture<String> get(String apiVersion, String path) {
```
Why is api version separate from base url?
If we add a new api version on the server, the mod should completely switch to the new api, right?

maxanier commented on src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java:
```diff
@@ -0,0 +1,143 @@
+        }
+        if (VampirismMod.inDev || settings == null) {
+            try {
+                InputStream inputStream = VampirismMod.class.getResourceAsStream("/config_values.json");
```
Maybe name file `default_remote_config`  or `fallback_api_values`. Also IOException text below is copy pasta

maxanier reviewed (commented):
Nice improvement

Cheaterpaul committed:
apply suggested changes to settings api

- settings and supporter only have optional return types
- rename non api methods
- remove api version as extra attribute
- rename config_values.json to default_remote_config.json
